src/Entities.php

```diff
@@ -132,10 +132,63 @@ protected function normalizeNames(Registry $registry): Registry
                     $r->setTarget($this->resolveTarget($registry, $r->getTarget()));
 
                     if ($r->getOptions()->has('though')) {
-                        $r->getOptions()->set(
-                            'though',
-                            $this->resolveTarget($registry, $r->getOptions()->get('though'))
-                        );
+                        $though = $r->getOptions()->get('though');
+                        if ($though !== null) {
+                            $r->getOptions()->set(
+                                'though',
+                                $this->resolveTarget($registry, $though)
+                            );
+                        }
+                    }
+
+                    if ($r->getOptions()->has('through')) {
+                        $through = $r->getOptions()->get('through');
+                        if ($through !== null) {
+                            $r->getOptions()->set(
+                                'through',
+                                $this->resolveTarget($registry, $through)
+                            );
+                        }
+                    }
+
+                    /**
+                     * We wanted to include composite keys support because it would be harder
+                     * to change the annotation signature later, but cycle itself doesn't support
+                     * composite keys on the ORM schema level, so we include a temporary hack
+                     * that limits the THROUGH inner and outer keys to a single element array.
+                     *
+                     * @see https://github.com/cycle/schema-builder/issues/25
```

Comment:
I think the `annotated` package should not have this restriction.

Reply:
Why not? Sometimes, it can get confusing whether the issue is with annotated schemas or data structure in general.

Reply:
The restriction is temporary and is supposed to be lifted once composite keys support is released.